Updated to Gtk 3.12

mikkeljohnsen opened the pull request:
Could this be merged to mono:master ?

Commit:
Mark deprecated functions as Obsolete

Commit:
Generator problems in GtkWidgetClass

Commit:
Updated to Gtk 3.12

Commit:
Make samples able to compile

Commit:
Include cairo in gapi

Commit:
Update to cairo API 1.12

Commit:
Int size on 64 bit

Commit:
Ignore gui-thread-check on windows

Commit:
Remove utf8 variant. Not used anymore on windows

Commit:
Fix conditional in Makefile.am

Commit:
Install cairo-api.xml

Commit:
Seperate pkgconfig version from package version

Commit:
Don't use API 3.12.0.0 just yet

mikkeljohnsen committed:
3.14.6

tesfabpel commented:
Hi, is there any roadmap for Gtk# 3?
Thanks :)

riccioclista committed:
csproj ToolsVersion to 4.0

nyxiscoo1 committed:
Removed CssProvider reference to unavailable interface GtkStyleProviderPrivate

mikkeljohnsen committed:
Merge pull request #1 from nyxiscoo1/master

Removed CssProvider reference to unavailable interface

Commit:
Merge remote-tracking branch 'upstream/master'

martinKupec committed:
Fix properties setting when constructing GLib.Objects

This is a proper fix of old FIXME.
It defers setting of managed properties during GLib.Object construction
until we have paired Managed and Unmanaged objects.

martinKupec committed:
Add possibility to pick glib type name for custom classes

This patch adds TypeName attribute which can be used on custom class
derived from GLib.Object in order to set GLib type name of the class.

Commit:
Track 3.16.6

Commit:
Fix IScrollable and List class

Commit:
Rename RowActivated to ListRowActivated for ListBox

Commit:
Return false if no border set

Commit:
Constructor

Commit:
Update csproj TargetFrameworkVersion

Commit:
Update References file's in csproj

Commit:
Update Solution to Visual Studio 2013

Commit:
Create gtkdotnet Project in Solution

Commit:
Revert: Seperate pkgconfig version from package version

Harol-Reina committed:
Changed type of property BackgroundGdk in gtk-api.raw

Harol-Reina committed:
build: Post-release version bump to 3.14.0, Update API_VERSION, Update PACKAGE_VERSION

mikkeljohnsen committed:
Merge pull request #2 from orion75/gtk-sharp-3-14-branch

Gtk sharp 3 14 branch

Harol-Reina committed:
gtk: Fix return types for null_term_array* methods in AboutDialog
this corrects the properties Authors, Artists and Documenters

Commit:
Update to Gtk 3.18.2

Had to change gapi2xml.pl to use the new properties method/macro in gtkwindow.c
- Issue, "invisible-chars" and "inner-border" is generated twice in gtkentry.c

Made patch to gwin32registrykey to use little endian by default. So Windows om ARM may fail.

mikkeljohnsen committed:
Use PACKAGE_VERSION 3.0

mikkeljohnsen committed:
Updated to gtk+ 3.18.5

mikkeljohnsen committed:
Update to 3.18.5. Downgrade API tp 3.0.0.0. So that you don't have to change Mono Developer project everytime you update gtk-sharp. We should update API to 4.0.0.0 when gtk+-4.0.0 is out.

mikkeljohnsen committed:
Merge pull request #3 from orion75/gtk-sharp-3-14-branch

gtk: Fix return types for null_term_array* methods in AboutDialog

cra0zy commented:
@bl8 @stsundermann Do you have any plans for Gtk Sharp 3 release? It's been several years now...

nyxiscoo1 committed:
Fixed memory leak when scheduling timeout and idle callbacks

mikkeljohnsen committed:
Merge pull request #7 from nyxiscoo1/mleakfix

Fixed memory leak when calling Application.Invoke and GLibSynchronizationContext from non GUI thread

chrislapa commented:
What is the status on this?

cra0zy commented:
@chrislapa If I may suggest something, don't bother using this repo for Gtk 3, instead use https://github.com/openmedicus/gtk-sharp, it's in much much better condition, that is it has Gtk 3.18 support :D.

harry-cpp committed:
Fix Dialog creation

mikkeljohnsen committed:
Merge pull request #8 from cra0zy/fixialogcreation

Fix Dialog creation

monojenkins commented:
Can one of the admins verify this patch?

monojenkins commented:
Can one of the admins verify this patch?

mikkeljohnsen committed:
Updated to Gtk+ 3.20.4

cra0zy commented:
If someone wants to use openmedicus version but doesn't want to compile it, I added nuget packages for it:
 - https://www.nuget.org/packages/gtk-sharp3/
 - https://www.nuget.org/packages/webkitgtk3-sharp/

Nuget sources:
 - https://github.com/cra0zy/gtk-sharp3-nuget
 - https://github.com/cra0zy/webkitgtk3-sharp-nuget

cra0zy commented:
@orion75 Your idea on using .targets file for copying .config files in NuGet package is a genius workaround :+1: 

I've been originally using you package for some time but you didn't update it for a while :(

mikkeljohnsen committed:
Made alias Symbol GdkRgba to GdkRGBA. This will provide properties of the type GdkRGBA in Gtk.

mikkeljohnsen committed:
Generate GdkColor properties

mikkeljohnsen committed:
Generate 'rgba' property

knocte commented:
> @orion75 Your idea on using .targets file for copying .config files in NuGet package is a genius workaround

workaround for what?

cra0zy commented:
> workaround for what?

Copying .config files to output dir...

knocte commented:
> Copying .config files to output dir...

Why is that relevant to this PR?

mikkeljohnsen committed:
Upstream patches

mikkeljohnsen committed:
Added StyleProviderPriority

Hecatron committed:
Small change to allow for building against latest gtk3

mikkeljohnsen committed:
Merge pull request #16 from grbd/master

Small change to allow for building against latest gtk3

mikkeljohnsen committed:
Updated to 3.20.6 and included SimpleAction

mikkeljohnsen committed:
Merge branch 'master' of github.com:openmedicus/gtk-sharp

mikkeljohnsen committed:
Merge from upstream

mikkeljohnsen committed:
Updated to 3.20.6

mikkeljohnsen committed:
Try to wrap NativeDialog for Windows

Hecatron committed:
Added python scripts for building NuGet packages via windows or linux

mikkeljohnsen committed:
Merge pull request #18 from grbd/Nuget

Added python scripts for building NuGet packages via windows or linux

Hecatron committed:
Rewrite of the nuget package genertion code, also included the glue libs

Hecatron committed:
Added some code for trying out the build of Gtksharp using dotnet cli from .Net core

Hecatron committed:
updated patches for 3.22

mikkeljohnsen committed:
Merge pull request #19 from ASoftTech/master

Pull request for a few different changes

grendello committed:
Fix System.Array usage in {List,Tree}Store

{List,Tree}Store.AppendValues overloads which take System.Array as their
'values' parameter incorrectly pass the array down to SetValues. The latter
expects a 'params object[]' array of parameters but passing an instance of
System.Array to such method will NOT pass the contents of System.Array instance
into the 'params' method as separate members of the parms array. It will instead
box System.Array and the params method will receive one parameter of type
System.Array instead of X parameters of various types. This affects the
following example code:

   var store = new TreeStore (typeof (string), typeof (int));
   store.AppendValues ("One", 1);

If the column configured to retrieve the 'string' value reads data from this
store it will instead get an instance of System.Array and the node displayed by
the TreeView will have no text, won't be clickable etc.

The fix implemented here is to "explode" the System.Array into a separate array
of the 'object[]' type.

The 'TreeStore.AppendValues (params object[] values)' overload no longer calls
the 'TreeStore.Appendvalues (Array values)' overload since the indirection only
wastes time and memory. It now directly calls `SetValues`.

grendello committed:
Fix the ActionMap.AddActionEntries API

The 'entries' parameter is an array.

mikkeljohnsen committed:
Merge pull request #21 from grendello/fix_actionmap

Fix the ActionMap.AddActionEntries API

mikkeljohnsen committed:
Merge pull request #20 from grendello/fix_array_usage

Fix System.Array usage in {List,Tree}Store

grendello committed:
Make GLib.Object.{Set,Get}Property public

The two methods are useful when we don't bind a property or the user
needs to set a custom property.

grendello committed:
Bind a number of Gtk.CellRendererTextProperties

This binds all the '*-set' properties. The properties are useful when one
wants to restore a previously set property to its initial value. For instance,
when setting the ForegroundRgba property the 'foreground-set' property is
automatically set to 'true' but when we assign 'Gdk.RGBA.Zero' to ForegroundRgba then the
'foreground-set' property is NOT set to false and the foreground becomes black instead
of the default one.

mikkeljohnsen committed:
Merge pull request #22 from grendello/object_setproperty_public

Make GLib.Object.{Set,Get}Property public

mikkeljohnsen committed:
Merge pull request #23 from grendello/text_rendered_properties

Bind a number of Gtk.CellRendererTextProperties

mikkeljohnsen committed:
Updated to 3.22. Need to fix GtkMenu and Popup

mikkeljohnsen committed:
Add Deprecated method Create in CairoHelper

mikkeljohnsen committed:
Updated version to 3.22.6

stil committed:
AppVeyor integration (#25)

* Fix minor bugs and improve code formatting of Python build tool

* Prepend $PATH to give it higher priority

* Invoke WiX tools through variable to ensure correct path

* Make build script ensure pacman dependencies are installed

* pacman -Sy causes dependency break on AppVeyor

* Skip deps version check and don't show progress bar

stil committed:
Include debugging symbols in nupkg (#26)

thiblahute committed:
Some fixes to support updated gstreamer-sharp (#29)

* parameters: Ignore UserData arguments to compute if has_optional

Those are never present in the prototypes and thus should not be taken
into account to see if prototype alternative has to be implemented.

* Fix generation of interface properties

* generator: Enhance the way we check if a field is padding

First check that fields are private to see if a field is padding,
in GstVideo we have public field staring with padding
(VideoAlignment.padding_top/bottomg/left/right) and thus need to be
taken into account to compare structures.

Also add a way for user to specify that the field is padding with
the new `is-padding` attribute.

In GStreamer padding fields usually look like __gst_reservedX, meaning
that they go unnoticed by previous code.

thiblahute committed:
Add meson build definitions (#30)

thiblahute committed:
meson: Add glue libraries generation (#31)

thiblahute committed:
meson: Install missing .pc/lib/gapi XML files (#32)

thiblahute committed:
meson: Specify all link dependencies in declare_dependency (#33)
